scheduler/src/cook/mesos/constraints.clj

```diff
@@ -160,10 +157,11 @@
       (let [vm-resources (.getCurrAvailableResources target-vm)
             vm-attributes (get-vm-lease-attr-map vm-resources)
             [passes? reason] (job-constraint-evaluate constraint vm-resources vm-attributes
-                                                      (into (vec (.getRunningTasks target-vm))
-                                                              (mapv (fn [^com.netflix.fenzo.TaskAssignmentResult result]
-                                                                      (.getRequest result))
-                                                                     (.getTasksCurrentlyAssigned target-vm))))]
+                                                      ; Use transients to improve speed and reduce allocations
```

Comment:
Consider this as a replacement to the current comment:

```
;; This function is called for each job considered, for each host available. It was optimized to reduce memory allocations because it accounted for ___% of all allocs 
```

I prefer comments to explain why something (weird) was done so new eyes (or forgetful eyes) don't need to wonder what compelled us to do the thing.